druid/src/widget/widget_ext.rs

```diff
@@ -159,6 +161,21 @@ pub trait WidgetExt<T: Data>: Widget<T> + Sized + 'static {
         ControllerHost::new(self, controller)
     }
 
+    /// Provide a closure that will be called when this widget is added to the widget tree.
+    ///
+    /// You can use this to perform any initial setup.
+    ///
+    /// This is equivalent to handling the [`LifeCycle::WidgetAdded`] event in a
+    /// custom [`Controller`].
+    ///
+    /// [`LifeCycle::WidgetAdded`]: crate::LifeCycle::WidgetAdded
+    fn on_added<W: Widget<T>>(
+        self,
+        f: impl Fn(&mut W, &mut LifeCycleCtx, &T, &Env) + 'static,
+    ) -> ControllerHost<Self, Added<T, W>> {
+        ControllerHost::new(self, Added::new(f))
+    }
```

Comment:
This is a bit of a gotcha, but I think instead of `W` we want to use `Self`, which refers to the type of whichever widget we're using this method on, like:

```suggestion
    fn on_added(
        self,
        f: impl Fn(&mut Self, &mut LifeCycleCtx, &T, &Env) + 'static,
    ) -> ControllerHost<Self, Added<T, Self>> {
        ControllerHost::new(self, Added::new(f))
    }
```

I think that's the only remaining fixup, and then we can get this merged. Thanks!